src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -852,12 +891,44 @@ def update_policy_definition(policy_definition_name, rules=None, params=None,
     return policy_client.policy_definitions.create_or_update(policy_definition_name, parameters)
 
 
+def update_policy_setdefinition(policy_set_definition_name, definitions=None, params=None,
+                                display_name=None, description=None):
+    if definitions:
+        if os.path.exists(definitions):
+            definitions = get_file_json(definitions)
+        else:
+            definitions = shell_safe_json_parse(definitions)
```

Comment:
If you just accept JSON and point of the @{file} syntax, you can eliminate this source of potential bugs.

Reply:
Reading your above comment I understand the reason to keep these commands similar, so this is fine.